blockchain/chainio.go

```diff
@@ -1387,7 +1387,7 @@ func (b *BlockChain) initChainState(ctx context.Context) error {
 	// block database info is loaded, but before block database migrations are
 	// run, since setting the initial UTXO set version depends on the block
 	// database version as that is where it originally resided.
-	if err := b.initUtxoDbInfo(ctx); err != nil {
+	if err := b.utxoCache.InitBackendInfo(b.dbInfo.version); err != nil {
```

Comment:
Similarly, this feels a little out of place to me.  I would generally expect the cache to handle whatever it needs to initialize **or** take an already initialized backend.  This seems like a bit of a weird mix of both which effectively tightly couples the cache to a specific style of backend.

Reply:
As mentioned in the other comment, I updated `BlockChain` to have a reference to the `UtxoBackend`, so it now calls `InitInfo` directly on the backend here.

Agreed that it is less than ideal.  The main reason the initialization of the backend info is inserted here, and the backend can't just initialize it on its own, is because of the migration path from before the database was separated (basically, as the comment in the above code mentions, it needs to know what version the block database before migrations are run to properly initialize the UTXO backend info).

Reply:
This is tied in with the aforementioned comment, so I'll resolve this one and we can continue the discussion there.